spoc/io.py

```diff
@@ -32,50 +35,29 @@ def load_label_library(path: str):
             label_library = pickle.load(handle)
         return label_library
 
-    def load_porec_fragments(self, path: str):
-        """Load porec fragments"""
-        data = self._parquet_reader_func(path)
-        if self._verify_schemas:
-            return FragmentSchema.validate(data)
-        return data
-
-    def load_annotated_fragments(self, path: str):
+    def load_fragments(self, path: str):
         """Load annotated fragments"""
         data = self._parquet_reader_func(path)
-        if self._verify_schemas:
-            return AnnotatedFragmentSchema.validate(data)
-        return data
+        return Fragments(data)
 
     @staticmethod
-    def write_annotated_fragments(
-        path: str, data: Union[pd.DataFrame, dd.DataFrame]
-    ) -> None:
+    def write_fragments(path: str, fragments: Fragments) -> None:
         """Write annotated fragments"""
-        data.to_parquet(path)
+        fragments.data.to_parquet(path)
 
     @staticmethod
-    def write_multiway_contacts(
-        path: str, data: Union[pd.DataFrame, dd.DataFrame]
-    ) -> None:
+    def write_multiway_contacts(path: str, contacts: Contacts) -> None:
         """Write multiway contacts"""
-        data.to_parquet(path)
+        contacts.data.to_parquet(path)
 
-    def load_multiway_contacts(
-        self, path: str, number_fragments: Union[int, None] = None
-    ) -> Union[pd.DataFrame, dd.DataFrame]:
+
+    # TODO; find a solution to hold information about symmetry flipping, label_sorting and binary_labeels equating
+    # -> do it with metadata of parquet file
+    def load_contacts(self, path: str, *args, **kwargs) -> Contacts:
         """Load multiway contacts"""
-        data = self._parquet_reader_func(path)
-        if self._verify_schemas:
-            if number_fragments is None:
-                raise ValueError(
-                    "Number of fragments needs to be specified if schema is validated!"
-                )
-            schema = HigherOrderContactSchema(number_fragments)
-            schema.validate_header(
-                data
-            )  # this is needed to catch problems that cause task graph construction failures
-            return schema.validate(data)
-        return data
+        return Contacts(
+            self._parquet_reader_func(path), *args, **kwargs
+        )
 
     @staticmethod
     def load_chromosome_sizes(path: str):
```

Comment:
Could we maybe wrap this around `bioframe.fetch_chromsizes`? It encapsulates both files and ucsc urls.

Reply:
Hmm, it seems like the interface of `bioframe.fecth_chromsizes`  is rather convoluted at first glance. Is there an easy way to read this from a file?

Reply:
I think that in the latest release (0.4.1) `bioframe.fetch_chromsizes` reads from a file by default, but one can simply change the behaviour (via `provider="ucsc"` or `"local"`). In bioframe 0.3.3 it probably can't read from the local storage.

Reply:
Ah, looking at the code, it should fetch chromsizes from a local file even in 0.3.3, just specifying `provider="local"`.